### ISSUE TITLE
ChainedMethodInvocationRefactor should be able to handle chains of arbitrary length

### ISSUE THREAD

**Issue by RadikalJin** [alfasoftware/astra#36]
**Describe the bug**
The `ChainedMethodInvocationRefactor `will currently match on two chained method invocations, and swap them to a different method.

For example,
`getCurrentFoo().doFooThing()`
can be swapped to
`doBarThing()`

**Expected behavior**
This refactor should be updated so that:
-  Chains of arbitrary length can be matched
- Those chains can themselves be replaced by other chains of arbitrary length, rather than just one method invocation.

**Additional context**
See Javadoc on `ChainedMethodInvocationRefactor` and associated tests for further example usage.

**Comment by monjurmorshed793** [comment]
A test case may be provided for understanding the issue more clearly.

**Comment by RadikalJin** [comment]
Good idea - added `TestMethodInvocationRefactor.testInvocationChainedWithLargeMatchingAndReplacementChains`.
